app/models/image.rb

```diff
@@ -15,6 +15,11 @@ def url(*args)
     image_data.file_url(*args)
   end
 
+  def filename_is_unique?
+    filename = image_data.file&.identifier
+    edition.images.none? { |i| i.image_data.carrierwave_image == filename }
+  end
```

Comment:
I think there might be some unexpected behaviour in here.

If we take an existing image from the database, `#filename_is_unique?` _will always be_ `false`. Even if it's the only image on that edition.

That's because `edition.images` will include the image we're working with, since it's already saved to the database. So it'll always match against itself.

For example:

```ruby
edition = Edition.last

# assuming an Edition with 1 image
edition.images.count # => 1

edition.images.first.filename_is_unique? # => false

# and that's because edition.images contains edition.images.first
edition.images.include?(edition.images.first) # => true

# and so there will never be 'none'
edition.images.none? { |i| i.image_data.carrierwave_image == filename } # => always false
```

This isn't a problem in our particular use case (i.e. in the edition images controller) because we call the `#filename_is_unique?` method _before we've persisted_ the new image to the database. So it won't be included in `edition.images`.

However I'd suggest we consider making this method more true to its name, and make it ignore itself from the list of images. For example, to extend the existing approach:

```suggestion
  def filename_is_unique?
    filename = image_data.file&.identifier
    edition.images.excluding(self).none? { |i| i.image_data.carrierwave_image == filename }
  end
```

(of course, we'd want to make sure we've covered this scenario in a unit test)